packages/samples/stencil/src/components/sample-component/sample-component.tsx

```diff
@@ -0,0 +1,123 @@
+import {Bindings, initializeBindings} from '@coveo/atomic';
+import {Component, Element, h, State, forceUpdate} from '@stencil/core';
+import {
+  PagerState,
+  Pager,
+  SearchStatusState,
+  buildSearchStatus,
+  buildPager,
+  Unsubscribe,
+} from '@coveo/atomic/headless';
+
+@Component({
+  tag: 'sample-component',
+  styleUrl: 'sample-component.css',
+  shadow: true,
+})
+export class SampleComponent {
+  private bindings?: Bindings;
+  private error?: Error;
+  private pagerController!: Pager;
+
+  private pagerUnsubscribe: Unsubscribe = () => {};
+  private statusUnsubscribe: Unsubscribe = () => {};
+  private i18nUnsubscribe = () => {};
+
+  @Element() private host!: Element;
+
+  @State() private pagerState!: PagerState;
+  @State() private statusState!: SearchStatusState;
+
+  public async connectedCallback() {
+    try {
+      this.bindings = await initializeBindings(this.host);
```

Comment:
out of curiosity, why don't we export the decorators for this?

Reply:
The decorators `InitializeBindings()` (e.g. do a lot of work under the hood. If we export them users might wonder what's happening with other lifecycle methods, they'll have to look at the implementation to understand. Better to have a single method.

Additionally, the decorators would only work for Stencil projects, and we prefer having one way for all frameworks. Also the decorators are Typescript (pre compilation) and we bundle Javascript in the project.